### DISCUSSION TITLE
Feasibility of porting Stylus to Manifest v3?

### DISCUSSION THREAD

**Discussion by chocmake** [openstyles/stylus#1761] [Q&A] [answered]
There's an open issue of migrating in #1430 and the deadline seems to be June 2024 when Chrome (unsure if affects Chromium, too) will begin disabling Manifest v2 extensions.

However seeing a comment mentioning uBlock Origin Lite and its [lack of cosmetic element picker](https://github.com/uBlockOrigin/uBlock-issues/issues/338#issuecomment-1507539114) feature or custom filters I became a bit concerned as to what Stylus features would be possible, given page manipulation is so essential to its purpose.

Anyone more familiar that could shed some light on this?

**Comment by tophf** [comment, marked as answer]
The only affected features are `Instant inject mode` and `Patch CSP to allow style assets`. There are no good workarounds yet, which is why I don't rush the migration. The migration itself isn't complicated.

There'll be also an annoyance with handling .user.css redirection to install the styles. In ManifestV3 we can't use webRequestBlocking to redirect it to our page, so the only way to do it in the same seamless fashion would be to use declarativeNetRequest + web_accessible_resources, which I'm reluctant to use as it makes the extension detectable even on sites that aren't styled. On the other hand, it's not really a big deal generally since we expose our styles in DOM on affected pages, so I'll probably use it and hope that Chrome finally implements use_dynamic_url this year (their promise to do it before removing MV2 is now about to become yet another lie out of the many already spouted by the MV3 team).

**Reply by dazzletune** [reply]
Thank you for your response. As of my comment, it is May 11th, 2024, and Google still plans to remove support for Manifest v2 extensions in Chromium-based browsers in June 2024, which is next month. If the Stylus developers do not release a Manifest v3 version of Stylus by June 2024, Sylus extension users will not be able to use Stylus anymore until there is a Manifest v3 version.

**Reply by tophf** [reply]
Per the announcement MV2 will be disabled in beta/dev/canary in June, not in the stable Chrome.

**Reply by dazzletune** [reply]
> Per the announcement MV2 will be disabled in beta/dev/canary in June, not in the stable Chrome.

True, but that only gives the Stylus developers about 4 to 6 weeks to create and publish a Manifest v3 version of Stylus. Google recommends that developers release Manifest v3 versions of their extensions prior to June 2024.

**Comment by rvcjew** [comment]
Any update to this now that it's been months as they have finally started saying it will come soon, have canary users been able to still run this extension just with missing features?

**Reply by tophf** [reply]
I'm working on it in #1828

**Reply by rvcjew** [reply]
Awesome ty. This extension is a life saver. Where can I donate to the devolpment?
